src/Hook/Input/PrePush.php

```diff
@@ -55,13 +66,41 @@ private function createFromStdIn(array $stdIn): array
             [$localRef, $localHash, $remoteRef, $remoteHash] = explode(' ', trim($line));
 
             if (Util::isZeroHash($remoteHash)) {
-                continue;
+                $remoteHash = $this->getHashOfBranchOrigin();
+
+                if (Util::isZeroHash($remoteHash)) {
+                    continue;
+                }
             }
 
-            $from     = new Ref($remoteRef, $remoteHash, Util::extractBranchFromRefPath($remoteRef));
-            $to       = new Ref($localRef, $localHash, Util::extractBranchFromRefPath($localRef));
+            $from = new Ref($remoteRef, $remoteHash, Util::extractBranchFromRefPath($remoteRef));
+            $to = new Ref($localRef, $localHash, Util::extractBranchFromRefPath($localRef));
             $ranges[] = new Range($from, $to);
         }
+
         return $ranges;
     }
+
+    private function getHashOfBranchOrigin(): string
+    {
+        if (!isset($this->hashOfBranchOrigin)) {
+            $currentBranch = $this->repository->getInfoOperator()
+                                              ->getCurrentBranch();
+
+            $processor = new Processor();
+            $reflog = $processor->run(sprintf('git reflog show --no-abbrev %s', $currentBranch))
```

Comment:
This should probably be moved to `SebastianFeldmann\Git\Repository` or one of the operator classes